src/backend/distributed/utils/colocation_utils.c

```diff
@@ -988,7 +988,6 @@ DeleteColocationGroup(uint32 colocationId)
 	{
 		simple_heap_delete(pgDistColocation, &(heapTuple->t_self));
 
-		CatalogUpdateIndexes(pgDistColocation, heapTuple);
```

Comment:
That's right. CatalogUpdateIndexes creates a new index pointer to the tuple, and that's not needed for a delete.